libp2p/multiaddress.nim

```diff
@@ -363,6 +402,10 @@ const
       mcodec: multiCodec("onion"), kind: Fixed, size: 10,
```

Comment:
Can you fix this to 12 while we're at it?

Reply:
Thought about that, but is it fine to do that in this PR?

Reply:
Yes go ahead, better than doing a PR for 1 LoC imo

Reply:
Maybe tests for the fix should also be added?

Reply:
Yes good point, maybe we should do another PR with:
- The fix
- Roundtrip tests (need to replace `ipfs` by `p2p` in the test inputs)
- Roundtrip tests "per part": `ma[0..^1].tryGet().mapIt($it).join('')` or something like that
The last one would have caught the bug in the first place